tests/test_tools.py

```diff
@@ -55,6 +55,19 @@ async def invalid_tool(ctx: RunContext[None]) -> str:  # pragma: no cover
     )
 
 
+def test_builtin_tool_registration():
+    """
+    Test that built-in functions can be registered as tools.
```

Comment:
We're really testing that they _can't_ be registered as tools, right, and that the user gets nice error message?